chore: clean up bash scripts

jniles opened the pull request:
This commit removes unused or deprecated scripts in the BHIMA repository and also improves some of the other bash scripts after passing through shellcheck linting.  This should relieve some maintenance burden on the developers.

Maintenance scripts (such as the install script, etc) should be in a separate repository third-culture-software/bhima-scripts.

To test this, just run the regular tests, as well as create a migration file (`npm run migrate`).  All should complete fully and without error.

jmcameron commented on sh/setup-migration-script.sh:
```diff
@@ -2,6 +2,14 @@
 
 # bash strict mode
 set -uo pipefail
+trap 'echo "Error: Command failed on line $LINENO"; exit 1' ERR
```
Nice!  👍

jmcameron commented on sh/test-e2e-all.sh:
```diff
@@ -22,33 +22,26 @@ function startfold {
 # run end to end account tests with Playwright
 for i in {1..8}; do
   date
-  startfold "Running account End to End Tests $i ...";
```
Since we are no longer doing "Folds", perhaps it makes sense to rename this function to something more obvious.

jniles replied:
Done.

jmcameron commented on sh/test-e2e-all.sh:
```diff
@@ -22,33 +22,26 @@ function startfold {
-then
-    echo "Deleting zombie node Bhima process $proc"
-    kill -9 $proc || true
+procs=$(netstat -tulpn |& grep 8080) || true
```
Just out of curiosity, why is "$(...)" preferred to "'...'" ?

jniles replied:
The answer applicable to this line is that it was done to satisfy a shell linter (https://www.shellcheck.net/).

[I've been told](https://stackoverflow.com/questions/9449778/what-is-the-benefit-of-using-instead-of-backticks-in-shell-scripts) in general that `$()` is preferred due to the ability to more easily nest commands, though that isn't applicable here.

jmcameron reviewed (approved):
The code changes look fine.
I had one minor comment about function names in the migration script.
All the tests pass.

jniles committed:
chore: clean up bash scripts

This commit removes unused or deprecated scripts in the BHIMA repository
and also improves some of the other bash scripts after passing through
shellcheck linting.  This should relieve some maintenance burden on the
developers.

Maintenance scripts (such as the install script, etc) should be in a
separate repository third-culture-software/bhima-scripts.
